crates/consensus/src/rounds/proposal.rs

```diff
@@ -92,14 +92,24 @@ impl ProposalState {
             return false
         };
 
-        let tx = TransactionRequest::default()
+        let mut tx = TransactionRequest::default()
             .with_to(handles.angstrom_address)
             .with_from(handles.signer.address())
             .with_input(bundle.pade_encode());
+
         let provider = handles.provider.clone();
+        let signer = handles.signer.clone();
 
         let submission_future = async move {
-            let submitted_tx = provider.send_transaction(tx).await.unwrap();
+            provider
+                .populate_gas_nonce_chain_id(signer.address(), &mut tx)
```

Comment:
The `.with_recommended_fillers()` for the provider builder includes the gas/nonce population.